src/sw.ts

```diff
@@ -1,28 +1,35 @@
 import mime from 'mime-types'
-import { getHelia } from './get-helia.ts'
+import { getVerifiedFetch } from './get-helia.ts'
 import { HeliaServiceWorkerCommsChannel, type ChannelMessage } from './lib/channel.ts'
 import { getSubdomainParts } from './lib/get-subdomain-parts.ts'
-import { heliaFetch } from './lib/heliaFetch.ts'
+import { getVerifiedFetchUrl, heliaFetch } from './lib/heliaFetch.ts'
 import { error, log, trace } from './lib/logger.ts'
 import { findOriginIsolationRedirect } from './lib/path-or-subdomain.ts'
-import type { Helia } from '@helia/interface'
+import type { VerifiedFetch } from '@helia/verified-fetch'
 
 declare let self: ServiceWorkerGlobalScope
 
-let helia: Helia
-self.addEventListener('install', () => {
+let verifiedFetch: VerifiedFetch
+
+self.addEventListener('install', (event) => {
+  // 👇 When a new version of the SW is installed, activate immediately
   void self.skipWaiting()
 })
 
 const channel = new HeliaServiceWorkerCommsChannel('SW')
 
 self.addEventListener('activate', () => {
+  // Set verified fetch initially
+  void getVerifiedFetch().then((newVerifiedFetch) => {
+    verifiedFetch = newVerifiedFetch
+  })
+
   channel.onmessagefrom('WINDOW', async (message: MessageEvent<ChannelMessage<'WINDOW', null>>) => {
     const { action } = message.data
     switch (action) {
       case 'RELOAD_CONFIG':
-        void getHelia().then((newHelia) => {
-          helia = newHelia
+        void getVerifiedFetch().then((newVerifiedFetch) => {
+          verifiedFetch = newVerifiedFetch
         })
```

Comment:
this is already async.. i wonder why I didn't use `verifiedFetch = await ...`

this should be updated

Reply:
fixed in https://github.com/ipfs-shipyard/helia-service-worker-gateway/pull/94